src/graph_engine.jl

```diff
@@ -258,11 +258,12 @@ getname(labels::ResizableArray{T, V, N} where {T <: NodeLabel, V, N}) = getname(
 iterate(label::NodeLabel) = (label, nothing)
 iterate(label::NodeLabel, any) = nothing
 
-to_symbol(label::NodeLabel) = Symbol(String(label.name) * "_" * string(label.global_counter))
+to_symbol(label::NodeLabel) = to_symbol(label.name, label.global_counter)
+to_symbol(name::D, index::Int) where {D} = Symbol(String(name) * "_" * string(index))
 
 Base.show(io::IO, label::NodeLabel) = print(io, label.name, "_", label.global_counter)
 Base.:(==)(label1::NodeLabel, label2::NodeLabel) = label1.name == label2.name && label1.global_counter == label2.global_counter
-Base.hash(label::NodeLabel, h::UInt) = hash(label.name, hash(label.global_counter, h))
+Base.hash(label::NodeLabel, h::UInt) = hash(label.global_counter, h)
```

Comment:
IMO it should actually be fine, `hash` can collide in principle. Also the documentation says
```
hash(x[, h::UInt]) -> UInt

  Compute an integer hash code such that isequal(x,y) implies
  hash(x)==hash(y). The optional second argument h is another hash code to be
  mixed with the result.
```
So in principle our `a == b` implies `hash(a) == hash(b)` and it follows the specification as documented. The opposite relation is not necessarily required.